tests/test_deserialization.py

```diff
@@ -1315,6 +1315,39 @@ class MySchema(Schema):
         errors = MySchema(partial=True).validate({'foo': 3}, partial=('bar', 'baz'))
         assert errors == {}
 
+    def test_unknown_fields_deserialization(self):
+        class MySchema(Schema):
+            foo = fields.Integer()
+
+        data = MySchema().load({'foo': 3, 'bar': 5})
+        assert data['foo'] == 3
+        assert 'bar' not in data
+
+        data = MySchema(unknown=ALLOW).load({'foo': 3, 'bar': 5}, unknown=IGNORE)
+        assert data['foo'] == 3
+        assert 'bar' not in data
+
+        data = MySchema().load({'foo': 3, 'bar': 5}, unknown=ALLOW)
+        assert data['foo'] == 3
+        assert data['bar']
+
+        data = MySchema(unknown=ALLOW).load({'foo': 3, 'bar': 5})
+        assert data['foo'] == 3
+        assert data['bar']
+
+        with pytest.raises(ValidationError) as excinfo:
+            MySchema(unknown=ALLOW).load({'foo': "asd", 'bar': 5})
+        assert 'foo' in str(excinfo)
+
+        schema = MySchema(unknown=ALLOW, many=True)
+        data = schema.load([{'foo': 3, 'bar': 5}])
+        assert 'foo' in data[0]
+        assert 'bar' in data[0]
+
+        with pytest.raises(ValidationError) as excinfo:
+            MySchema(unknown=RAISE).load({'foo': 3, 'bar': 5})
+        assert 'bar' in str(excinfo)
+
```

Comment:
The code isn't handling list deserialization. The following test fails:

```python
        with pytest.raises(ValidationError) as excinfo:
            MySchema(unknown=RAISE, many=True).load([
                {'foo': 'abc'},
                {'foo': 3, 'bar': 5},
            ])
        err = excinfo.value
        assert 0 in err.messages
        assert 'foo' in err.messages[0]
        assert err.messages[0]['foo'] == ['Not a valid integer.']
        assert 1 in err.messages
        assert 'bar' in err.messages[1]
        assert err.messages[1]['bar'] == ['Unknown key.']
```

```
|| ______________________________________________________________________ TestSchemaDeserialization.test_unknown_fields_deserialization _______________________________________________________________________
|| tests/test_deserialization.py:1360: in test_unknown_fields_deserialization
||     assert 1 in err.messages
|| E   AssertionError: assert 1 in {0: {'foo': ['Not a valid integer.']}, 'bar': ['Unkown key.']}
|| E    +  where {0: {'foo': ['Not a valid integer.']}, 'bar': ['Unkown key.']} = ValidationError({0: {'foo': ['Not a valid integer.']}, 'bar': ['Unkown key.']},).messages
|| ========================================================================================= 1 failed in 0.11 seconds =========================================================================================
```

Reply:
Fixed, thanks.